lib/internal/test_runner/mock/fake_timers.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const {
+  DateNow,
+  FunctionPrototypeBind,
+  globalThis,
+  Promise,
+} = primordials;
+
+const PriorityQueue = require('internal/priority_queue');
+const nodeTimers = require('timers');
+const nodeTimersPromises = require('timers/promises');
+
+function compareTimersLists(a, b) {
+  return (a.runAt - b.runAt) || (a.id - b.id);
+}
+
+function setPosition(node, pos) {
+  node.priorityQueuePosition = pos;
+}
+
+
+class FakeTimers {
+  #realSetTimeout;
+  #realClearTimeout;
+  #realSetInterval;
+  #realClearInterval;
+
+  #isEnabled = false;
+  #currentTimer = 1;
+  #fakeTimers = {};
+  #now = DateNow();
+
+  #timers = new PriorityQueue(compareTimersLists, setPosition);
+
+  #setTimeout = FunctionPrototypeBind(this.#createTimer, this, false);
+  #clearTimeout = FunctionPrototypeBind(this.#clearTimer, this);
+  #setInterval = FunctionPrototypeBind(this.#createTimer, this, true);
+  #clearInterval = FunctionPrototypeBind(this.#clearTimer, this);
+
+  #createTimer(isInterval, callback, delay, ...args) {
+    const timerId = this.#currentTimer++;
+    this.#timers.insert({
+      id: timerId,
+      callback,
+      runAt: DateNow() + delay,
+      interval: isInterval,
+      args,
+    });
+
+    return timerId;
+  }
+
+  #clearTimer(position) {
+    this.#timers.removeAt(position);
+  }
+
+  tick(time = 1) {
+
+    // if (!this.isEnabled) {
+    //   throw new Error('you should enable fakeTimers first by calling the .enable function');
+    // }
+    this.#now += time;
+    const timers = this.#timers;
+
+    for (let timer = timers.peek(); timer?.runAt <= this.#now; timer = timers.peek()) {
+      timer.callback(...timer.args);
+      timers.shift();
+      if (timer.interval) {
+        timer.runAt = timer.runAt + timer.interval;
+        timers.insert(timer);
+      }
+    }
+
+  }
+
+  enable() {
+    // if (this.isEnabled) {
+    //   throw new Error('fakeTimers is already enabled!');
+    // }
+    this.#now = DateNow();
+    this.#isEnabled = true;
+
+    this.#realSetTimeout = globalThis.setTimeout;
+    this.#realClearTimeout = globalThis.clearTimeout;
+    this.#realSetInterval = globalThis.setInterval;
+    this.#realClearInterval = globalThis.clearInterval;
+
+    globalThis.setTimeout = this.#setTimeout;
+    globalThis.clearTimeout = this.#clearTimeout;
+    globalThis.setInterval = this.#setInterval;
+    globalThis.clearInterval = this.#clearInterval;
+
+
+    nodeTimers.setTimeout = this.#setTimeout;
+    nodeTimers.clearTimeout = this.#clearTimeout;
+    nodeTimers.setInterval = this.#setInterval;
+    nodeTimers.clearInterval = this.#clearInterval;
```

Comment:
why are these being set here? They're not being cleared in `reset`, so it seems like they could just be public fields?

Reply:
they're not yet! haha I haven't finished working on the reset method. But they'll be also there